core/src/com/unciv/logic/automation/civilization/NextTurnAutomation.kt

```diff
@@ -801,6 +804,29 @@ object NextTurnAutomation {
         }
     }
 
+    private fun offerDefensivePact(civInfo: Civilization) {
+        if (!civInfo.diplomacyFunctions.canSignDefensivePact()) return // don't waste your time
+
+        val canSignDefensivePactCiv = civInfo.getKnownCivs()
+            .filter {
+                civInfo.diplomacyFunctions.canSignDefensivePactWith(it)
+                    && !civInfo.getDiplomacyManager(it).hasFlag(DiplomacyFlags.DeclinedDefensivePact)
+                    && civInfo.getDiplomacyManager(it).relationshipIgnoreAfraid() == RelationshipLevel.Ally
+            }
+            .sortedByDescending { it.stats.statsForNextTurn.science }
+
+        for (otherCiv in canSignDefensivePactCiv) {
+            // Default setting is 1, this will be changed according to different civ.
+            if ((1..10).random() > 1) continue
```

Comment:
So 10% chance? Random.nextInt(10) is the way to do this, unless range has its own random() then we're building a list here for no reason

Reply:
Ok cool, then not a problem 👍

Reply:
The line above it in offerDeclarationOfFriendship() at 780 has: if ((1..10).random() <= 5), so I just copied that. Should I replace both with Random.nextInt(10)?

Reply:
I will leave it out of this pull request because it is consistent with offerResearchAgreement() and offerDeclarationOfFriendship. We can change those in a different pull request if we want.